jdaviz/configs/mosviz/tests/test_data_loading.py

```diff
@@ -0,0 +1,225 @@
+
+    table = mosviz_helper.app.get_viewer('table-viewer')
+    table.widget_table.vue_on_row_clicked(0)
+
+    data = mosviz_helper.app.get_data_from_viewer('image-viewer')
+
+    assert isinstance(list(data.values())[0], CCDData)
+    assert list(data.values())[0].shape == (55, 55)
+    assert list(data.keys())[0] == f"{label} 0"
+
+
+def test_load_spectrum_collection(mosviz_helper, spectrum_collection):
+    labels = [f"Test Spectrum Collection {i}" for i in range(5)]
+    mosviz_helper.load_1d_spectra(spectrum_collection, data_labels=labels)
+
+    assert len(mosviz_helper.app.data_collection) == 6
+    assert mosviz_helper.app.data_collection[0].label == labels[0]
+
+    table = mosviz_helper.app.get_viewer('table-viewer')
+    table.widget_table.vue_on_row_clicked(0)
+
+    data = mosviz_helper.app.get_data_from_viewer('spectrum-viewer')
+
+    assert isinstance(list(data.values())[0], Spectrum1D)
+    assert list(data.keys())[0] == labels[0]
+
+
+def test_load_list_of_spectrum1d(mosviz_helper, spectrum1d):
+    spectra = [spectrum1d]*3
+
+    labels = [f"Test Spectrum 1D {i}" for i in range(3)]
+    mosviz_helper.load_1d_spectra(spectra, data_labels=labels)
+
+    assert len(mosviz_helper.app.data_collection) == 4
+    assert mosviz_helper.app.data_collection[0].label == labels[0]
+
+    table = mosviz_helper.app.get_viewer('table-viewer')
+    table.widget_table.vue_on_row_clicked(0)
+
+    data = mosviz_helper.app.get_data_from_viewer('spectrum-viewer')
+
+    assert isinstance(list(data.values())[0], Spectrum1D)
+    assert list(data.keys())[0] == labels[0]
+
+
+@pytest.mark.filterwarnings('ignore')
+def test_load_mos_spectrum2d(mosviz_helper, mos_spectrum2d):
+
+    label = "Test 2D Spectrum"
+    mosviz_helper.load_2d_spectra(mos_spectrum2d, data_labels=label)
+
+    assert len(mosviz_helper.app.data_collection) == 2
+    assert mosviz_helper.app.data_collection[0].label == label
+
+    table = mosviz_helper.app.get_viewer('table-viewer')
+    table.widget_table.vue_on_row_clicked(0)
+
+    data = mosviz_helper.app.get_data_from_viewer('spectrum-2d-viewer')
+
+    assert list(data.values())[0].shape == (1024, 15)
+    assert list(data.keys())[0] == label
+
+
+@pytest.mark.filterwarnings('ignore')
+@pytest.mark.parametrize('label', [None, "Test Label"])
+def test_load_multi_image_spec(mosviz_helper, mos_image, spectrum1d, mos_spectrum2d, label):
+    spectra1d = [spectrum1d]*3
+    spectra2d = [mos_spectrum2d]*3
+    images = [mos_image]*3
+
+    mosviz_helper.load_data(spectra1d, spectra2d, images=images, images_label=label)
+
+    assert mosviz_helper.app.get_viewer("table-viewer").figure_widget.highlighted == 0
+    assert len(mosviz_helper.app.data_collection) == 10
+
+    qtable = mosviz_helper.to_table()
+    if label is None:
+        assert qtable["Images"][0] == "Image 0"
+    else:
+        assert qtable["Images"][0] == "Test Label 0"
+
+
+@pytest.mark.filterwarnings('ignore')
```

Comment:
Same comment as "is it possible to create a test case that is warning free?" above.